ClassicShellSrc/BUILDME.txt

```diff
@@ -5,10 +5,13 @@ for other languages.
 The final files (installers, archives) are saved to the ClassicShellSetup\Final folder.
 
 You need the following tools:
-Visual Studio 2008
-Windows SDK 7.1
+Visual Studio 2017 (Community Edition is enough)
+  - Desktop development with C++ workload
+  - Windows 10 SDK (10.0.16299.0) for Desktop C++
+  - Visual C++ ATL support
 HTML Help Workshop
 WiX 3.7
+WinRAR
```

Comment:
7z ?

Reply:
Build scripts use WinRAR to create some archives. I've just _documented_ this fact.
Though I agree it may be good idea to use some free archiving utility.

Reply:
@ge0rdi @Quentinix Yes. There's many 7z/LZMA2 stuff out there, it can be integrated with WiX and there are many InstallShield forks with it. Are you still interested on this? I'm at very early programming learning steps, I even need to prepare my systems and toolchains (and get more space for the Visual Studio I want to install, it's BIG).

Please consider it. Windows needs more Open Source, less proprietary software. 

I dream with a fully functional, extremely efficient, secure, robust, extreme hardware support and extremely compatible [ReactOS](https://reactos.org). I know it's a [Pipe Dream](https://en.oxforddictionaries.com/definition/pipe_dream), but a man can dream...

Meanwhile, I would love to be able progressively be able to replace original Microsoft Windows components with 100% compatible Free Software equivalents. I think of it like replacing small parts of yourself with synthetic one: Would it be the same after all  parts are complete? ;)

Reply:
This PR is about migration to VS2017.
I'd suggest to create separate [issue](https://github.com/coddec/Classic-Shell/issues/new) for replacing WinRar with free alternative.

Reply:
I couldn't justify creating a new issue simply for this comment alone, but one of the reasons WinRAR is superior to 7z for many projects is that **the LZMA2 algorithm does not currently have a readily available, multi-threaded de/compression utility.** Single-threaded work on archives can be painfully slow.
Some old sources exist for cygwin/MSYS2-enabled versions of bzip2 but I could not get them to compile and they haven't been updated *for nearly a decade*.